### DISCUSSION TITLE
How to create a lua color scheme (source example / help tutorial?)

### DISCUSSION THREAD

**Discussion by shmerl** [neovim/neovim#28850] [Q&A]
Is there some comprehensive example of color scheme defined in lua? Default one is defined in highlight_group.c and not in lua.

Is https://github.com/neovim/neovim/blob/master/runtime/colors/vim.lua a good example to use for creating a new scheme / converting existing one from vimscript to lua?

**Comment by adigitoleo** [comment]
In v0.10, some significant changes were made to highlight groups and the default colorscheme. Some of these changes are arguably not easy to find (yet) in the helpfiles. You should look in at least these places: `:h group-name`, `:h diagnostic-highlights`, `:h lsp-highlight` and `:h treesitter-highlight-groups`. The addition of lsp and treesitter groups means that some extra choices will need to be made when porting vim8 themes. Most of the work consists in finding which highlight groups are required in the minimal case. One way to get some idea about this is to do `nvim -u NORC` and then inspect the listing provided by `:hi`.

The file you linked to is also helpful, but note that it omits definition of `Normal`, and sets some other opinionated lsp and diagnostic highlight links and definitions. I would not treat it as a complete example of a minimally viable colorscheme.

See also https://github.com/neovim/neovim/issues/29013 for some rationale behind the changes.

**Comment by adigitoleo** [comment]
It isn't lua (I have users who are on vim8), but I have a fairly minimalist colorscheme [mellow](https://github.com/adigitoleo/vim-mellow) which may help as I've recently updated it to include what I found to be the minimal set of 0.10 groups. See the file `colors/mellow.vim`, in particular lines 87-341. Of course the disclaimer here is that I'm the author and did make some opinionated choices as well.
